app/scripts/modules/amazon/src/function/CreateLambdaFunction.tsx

```diff
@@ -0,0 +1,244 @@
+import * as React from 'react';
+import { cloneDeep } from 'lodash';
+import {
+  IFunctionModalProps,
+  FunctionWriter,
+  ReactInjector,
+  ReactModal,
+  TaskMonitor,
+  WizardModal,
+  WizardPage,
+  noop,
+  Application,
+} from '@spinnaker/core';
+
+import { IAmazonFunction, IAmazonFunctionUpsertCommand } from 'amazon/domain';
+import { FunctionBasicInformation } from './configure/FunctionBasicInformation';
+import { ExecutionRole } from './configure/ExecutionRole';
+import { FunctionSettings } from './configure/FunctionSettings';
+import { FunctionEnvironmentVariables } from './configure/FunctionEnvironmentVariables';
+import { Network } from './configure/Network';
+import { AwsFunctionTransformer } from './function.transformer';
+import { FunctionTags } from './configure/FunctionTags';
+import { FunctionDebugAndErrorHandling } from './configure/FunctionDebugAndErrorHandling';
+
+export interface IAmazonCreateFunctionProps extends IFunctionModalProps {
+  app: Application;
+  functionDef: IAmazonFunction;
+}
+export interface IAmazonCreateFunctionState {
+  isNew: boolean;
+  functionCommand: IAmazonFunctionUpsertCommand;
+  taskMonitor: TaskMonitor;
+}
+
+export class CreateLambdaFunction extends React.Component<IAmazonCreateFunctionProps, IAmazonCreateFunctionState> {
+  public static defaultProps: Partial<IAmazonCreateFunctionProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  constructor(props: IAmazonCreateFunctionProps) {
+    super(props);
+    const functionTransformer = new AwsFunctionTransformer();
+    const funcCommand = props.command
+      ? (props.command as IAmazonFunctionUpsertCommand) // ejecting from a wizard
+      : props.functionDef
+      ? functionTransformer.convertFunctionForEditing(props.functionDef)
+      : functionTransformer.constructNewAwsFunctionTemplate(props.app);
+
+    this.state = {
+      isNew: !props.functionDef,
+      functionCommand: funcCommand,
+      taskMonitor: null,
+    };
+  }
+
+  private _isUnmounted = false;
+  private refreshUnsubscribe: () => void;
```

Comment:
I feel like these variables should be part of the state. 

`_isUnmounted` is a tricky one.  If a task completes after a user closes the modal, does it matter if we truncate the logic in `onApplicationRefresh`? At that point it seems like the parent component should be keeping track of whether or not the component is open/closed, and that this flag would be better served as a prop since it offers insight about a higher level of the DOM.

Reply:
> Do we know if `_isUnmounted` is necessary? I don't think it it because `refreshUnsubscribe()` is being called in `cWU`.
> 
> Also I think `refreshUnsubscribe` is fine as an instance variable, as it's unrelated to `render()`.

`_isUnmounted` is also used in `onApplicationRefresh`

Reply:
oh I see it now